src/backends/gstreamer/audio_sink.rs

```diff
@@ -63,15 +40,12 @@ impl AudioSink for GStreamerAudioSink {
             .ok_or(())?;
         src.set_caps(&info.to_caps().unwrap());
         src.set_property_format(gst::Format::Time);
-        let settings = Settings::default();
         let mut sample_offset = 0;
-        let n_samples = settings.samples_per_buffer as u64;
-        let buf_size = (n_samples as usize) * (info.bpf() as usize);
         let rate = info.rate();
 
         let graph_ = graph.clone();
-        let need_data = move |app: &AppSrc, _bytes: u32| {
-            let mut buffer = gst::Buffer::with_size(buf_size).unwrap();
+        let need_data = move |app: &AppSrc, bytes: u32| {
+            let mut buffer = gst::Buffer::from_slice(&SILENCE[0..bytes as usize]).unwrap();
```

Comment:
Also you use the bytes as passed in by appsrc. In theory that might be more than 4096, in theory it might not be a multiple of the unit size (sample size * channels), and worst: the alignment of the silence `Vec` might not be correct (it needs to be aligned to the sample size). You probably want to allocate always the same buffer size here like before and not worry about anything :)